core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -299,46 +464,32 @@ public void reset() {
    * @return whether to process the update loop
    */
   public boolean preUpdate(float dt) {
-    InputController input = InputController.getInstance();
-    input.readInput(level.getBounds(), level.getScale());
+
+    gameInputController.readInput(level.getBounds(), level.getScale());
     if (listener == null) {
       return true;
     }
 
     // Toggle debug
-    if (input.didDebug()) {
+    if (gameInputController.didDebug()) {
       level.setDebug(!level.getDebug());
     }
 
     // Handle resets
-    if (input.didReset()) {
+    if (gameInputController.didReset()) {
       reset();
     }
-    if (input.getNextLevel()) {
-      if (levelNumber < levels.size() - 1) {
-        levelNumber += 1;
-        levelFormat = directory.getEntry(levels.get(levelNumber), JsonValue.class);
-        reset();
-        input.setNextLevel();
-      } else {
-        reset();
-        input.setNextLevel();
-      }
+    if (gameInputController.didNextLevel()) {
+      assert listener != null;
+      listener.exitScreen(this, EXIT_TO_NEXT_LEVEL);
     }
-    if (input.getPastLevel()) {
-      if (levelNumber != 0) {
-        levelNumber -= 1;
-        levelFormat = directory.getEntry(levels.get(levelNumber), JsonValue.class);
-        reset();
-        input.setPastLevel();
-      } else {
-        reset();
-        input.setPastLevel();
-      }
+    if (gameInputController.didPrevLevel()) {
+      assert listener != null;
+      listener.exitScreen(this, EXIT_TO_PREV_LEVEL);
     }
```

Comment:
I noticed that the previous and next level buttons are no longer working, it always skips level 1 and can only switch between level 0 and level 2. Since we have a level selector, I think it would be better to just remove this functionality for now instead of trying to debug it.

Reply:
Sounds good